tests/csapi/room_relations_test.go

```diff
@@ -205,3 +206,91 @@ func TestRelationsPagination(t *testing.T) {
 		},
 	})
 }
+
+func TestRelationsPaginationSync(t *testing.T) {
+  runtime.SkipIf(t, runtime.Dendrite) // FIXME: https://github.com/matrix-org/dendrite/issues/2944
+
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	roomID := alice.CreateRoom(t, map[string]interface{}{"preset": "public_chat"})
+	_, token := alice.MustSync(t, client.SyncReq{})
+
+	res := alice.MustDoFunc(t, "PUT", []string{"_matrix", "client", "v3", "rooms", roomID, "send", "m.room.message", "txn-1"}, client.WithJSONBody(t, map[string]interface{}{
```

Comment:
Nothing wrong with this, but you might prefer to use `SendEventUnsynced` here

https://github.com/matrix-org/complement/blob/70f58acc4576daddaba968a8a29db55b04418c5d/internal/client/client.go#L263-L276